packages/components/src/navigation/index.js

```diff
@@ -121,6 +121,96 @@ const Navigation = ( { activeItemId, children, data, rootTitle } ) => {
 	);
 };
 
+export function Navigation2( { children } ) {
+	const [ activeLevel, setActiveLevel ] = useState( 'root' );
```

Comment:
The initial `activeLevel` is going to be difficult to determine. If the go to a url with a deeply nested nav item, the level will need to be determined somehow. I'll need to think on this one.

Reply:
This is the only drawback I see to a more compositional route. We lose benefits for determining active state for levels and items.  We won't be able to use the experimental route detection I was working on in https://github.com/WordPress/gutenberg/pull/24985 which is okay, but as @psealock mentioned, this puts a lot of responsibility on the consumer to detect active levels and items.

Reply:
Actually, thinking about this a bit more, if we do decide to go with route detection, we could simply pass down a method or prop or even determine this directly in the `MenuItem`.  So that's a non-issue.

All that remains is to figure out how to set the active level/status on init and on change when many levels exist.